### ISSUE TITLE
[Bug]: Legends used for Charts have hardcoded role as link which leads to accessibility issue

### ISSUE THREAD

**Issue by kumaras16** [microsoft/fluentui#24692]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
npx: installed 1 in 3.495s

  System:
    OS: Windows 10 10.0.22000
    CPU: (16) x64 Intel(R) Xeon(R) W-2245 CPU @ 3.90GHz
    Memory: 22.40 GB / 63.60 GB
  Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (104.0.1293.54)
    Internet Explorer: 11.0.22000.120
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://github.com/microsoft/fluentui/blob/bfe1eb625184535fe3dbd83903eb03563814d28b/packages/charting/src/components/Legends/Legends.base.tsx#L307

## Actual Behavior
LineChart have legends present at the bottom. In case of Overflow legends are shown as a group with title as **{Number} more**. role for this html element is hardcoded as link which is an accessibility issue because it's not redirecting to another page.
https://github.com/microsoft/fluentui/blob/bfe1eb625184535fe3dbd83903eb03563814d28b/packages/charting/src/components/Legends/Legends.base.tsx#L307
## Expected Behavior
role here should be button as onHover of that button we are showing overflowing legends or it should take role as an parameter

### Requested priority

High

### Products/sites affected

Sharepoint admin center

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
Looping in @Raghurk to the issue for visibility.
